### DISCUSSION TITLE
opentelemetry::v1::exporter::otlp::OtlpHttpExporterOptions throws me stack-buffer-overflow error on addresssanitizer for version 1.12.0

### DISCUSSION THREAD

**Discussion by sauravslr** [open-telemetry/opentelemetry-cpp#2462] [Q&A]
i get stack buffer overflow error from addresssanitizer when using opentelemetry::v1::exporter::otlp::OtlpHttpExporterOptions to export spans to Jaeger UI. The exporter works for me in v1.10.0. the function almost looks like the example function as mentioned in the repository 

```cpp
void InitTracer()
{
	opentelemetry::sdk::resource::ResourceAttributes resource_attributes = {
		{"service.name", "Account-provider"},
		{"service.version", "1.1.0"}
	};
	auto resource = opentelemetry::sdk::resource::Resource::Create(resource_attributes);

	opentelemetry::exporter::otlp::OtlpHttpExporterOptions opts;
	opts.url = "http://localhost:4318/v1/traces";
	auto exporter = opentelemetry::exporter::otlp::OtlpHttpExporterFactory::Create(opts);
	auto processor =
		opentelemetry::sdk::trace::SimpleSpanProcessorFactory::Create(std::move(exporter));
	std::vector<std::unique_ptr<opentelemetry::sdk::trace::SpanProcessor>> processors;
	processors.push_back(std::move(processor));
	// Default is an always-on sampler.
	std::unique_ptr<opentelemetry::sdk::trace::TracerContext> context =
		opentelemetry::sdk::trace::TracerContextFactory::Create(std::move(processors), resource);
	std::shared_ptr<opentelemetry::trace::TracerProvider> provider =
		opentelemetry::sdk::trace::TracerProviderFactory::Create(std::move(context));
	// Set the global trace provider
	opentelemetry::trace::Provider::SetTracerProvider(provider);


}

**Comment by lalitb** [comment]
Can you share more details - stack dump or actual error?

**Reply by sauravslr** [reply]
![otlpexporter](https://github.com/open-telemetry/opentelemetry-cpp/assets/39926263/13fb39f3-6e40-4000-99a4-e15473bf9817)

**Reply by marcalff** [reply]
Please do not paste a screen shot.

Paste the text instead.

**Reply by sauravslr** [reply]
Sorry abou that. Pasting the text here `=================================================================
==18528==ERROR: AddressSanitizer: stack-buffer-overflow on address 0x00dd55afdaa0 at pc 0x7ff6ed343b9f bp 0x00dd55afd400 sp 0x00dd55afd408
WRITE of size 8 at 0x00dd55afdaa0 thread T0
==18528==WARNING: Failed to use and restart external symbolizer!
    #0 0x7ff6ed343b9e in std::_Container_base12::_Container_base12 C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.34.31933\include\xmemory:1080
    #1 0x7ff6ed3083ce in std::_String_val<std::_Simple_types<char> >::_String_val<std::_Simple_types<char> > C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.34.31933\include\xstring:2271
    #2 0x7ff6ed23fbd3 in std::_Compressed_pair<std::allocator<char>,std::_String_val<std::_Simple_types<char> >,1>::_Compressed_pair<std::allocator<char>,std::_String_val<std::_Simple_types<char> >,1><> C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.34.31933\include\xmemory:1376
    #3 0x7ff6ed3125e9 in std::basic_string<char,std::char_traits<char>,std::allocator<char> >::basic_string<char,std::char_traits<char>,std::allocator<char> > C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.34.31933\include\xstring:2615
    #4 0x7ff6f0654558 in opentelemetry::v1::exporter::otlp::OtlpHttpExporterOptions::OtlpHttpExporterOptions E:\.conan\772a6c\1\source\exporters\otlp\src\otlp_http_exporter_options.cc:18
    #5 0x7ff6ef0cc85a in tracing::InitTracer C:\Users\INT005702\master\Provider\AccountProvider\Tracer_Common.cpp:17
    #6 0x7ff6eec329c5 in tvaccount::AccountConfiguration::AccountConfiguration C:\Users\INT005702\master\Provider\AccountProvider\Configuration\AccountConfiguration.cpp:112
    #7 0x7ff6ed514daa in std::_Construct_in_place<tvaccount::AccountConfiguration,tvversion::BuildVersion> C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.34.31933\include\xutility:231
    #8 0x7ff6ed514ae0 in std::_Optional_construct_base<tvaccount::AccountConfiguration>::_Construct<tvversion::BuildVersion> C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.34.31933\include\optional:147
    #9 0x7ff6ed5167fe in std::optional<tvaccount::AccountConfiguration>::emplace<tvversion::BuildVersion> C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.34.31933\include\optional:333
    #10 0x7ff6ed512d2d in GlobalTestFixture::GlobalTestFixture C:\Users\INT005702\master\Provider\AccountTest\Fixtures\GlobalTestFixture.cpp:21
    #11 0x7ff6eebb884e in boost::unit_test::ut_detail::global_configuration_impl<GlobalTestFixture>::test_start C:\.conan\dd73cc\1\include\boost\test\tree\global_fixture.hpp:91
    #12 0x7ff6f1bbccac in boost::_mfi::mf2<void,boost::unit_test::test_observer,unsigned long,unsigned long>::operator() E:\.conan\2d8b53\1\src\boost\bind\mem_fn_template.hpp:280
    #13 0x7ff6f1baa2d4 in boost::_bi::list3<boost::_bi::value<boost::unit_test::test_observer *>,boost::_bi::value<boost::unit_test::test_case_counter::readonly_property43>,boost::_bi::value<unsigned long> >::operator()<boost::_mfi::mf2<void,boost::unit_test::test_observer,unsigned long,unsigned long>,boost::_bi::list0> E:\.conan\2d8b53\1\src\boost\bind\bind.hpp:378
    #14 0x7ff6f1bbc64a in boost::_bi::bind_t<void,boost::_mfi::mf2<void,boost::unit_test::test_observer,unsigned long,unsigned long>,boost::_bi::list3<boost::_bi::value<boost::unit_test::test_observer *>,boost::_bi::value<boost::unit_test::test_case_counter::readonly_property43>,boost::_bi::value<unsigned long> > >::operator() E:\.conan\2d8b53\1\src\boost\bind\bind.hpp:1274
    #15 0x7ff6f1bc96b2 in boost::detail::function::void_function_obj_invoker0<boost::_bi::bind_t<void,boost::_mfi::mf2<void,boost::unit_test::test_observer,unsigned long,unsigned long>,boost::_bi::list3<boost::_bi::value<boost::unit_test::test_observer *>,boost::_bi::value<boost::unit_test::test_case_counter::readonly_property43>,boost::_bi::value<unsigned long> > >,void>::invoke E:\.conan\2d8b53\1\src\boost\function\function_template.hpp:158
    #16 0x7ff6f1bbc728 in boost::function0<void>::operator() E:\.conan\2d8b53\1\src\boost\function\function_template.hpp:771
    #17 0x7ff6f1c20885 in boost::detail::forward::operator() E:\.conan\2d8b53\1\src\boost\test\impl\execution_monitor.ipp:1405
    #18 0x7ff6f1c210c2 in boost::detail::function::function_obj_invoker0<boost::detail::forward,int>::invoke E:\.conan\2d8b53\1\src\boost\function\function_template.hpp:137
    #19 0x7ff6f1c20868 in boost::function0<int>::operator() E:\.conan\2d8b53\1\src\boost\function\function_template.hpp:771
    #20 0x7ff6f1c1fea2 in boost::detail::do_invoke<boost::shared_ptr<boost::detail::translator_holder_base>,boost::function<int __cdecl(void)> > E:\.conan\2d8b53\1\src\boost\test\impl\execution_monitor.ipp:318
    #21 0x7ff6f1c1f157 in boost::execution_monitor::catch_signals E:\.conan\2d8b53\1\src\boost\test\impl\execution_monitor.ipp:1236
    #22 0x7ff6f1c1ef40 in boost::execution_monitor::execute E:\.conan\2d8b53\1\src\boost\test\impl\execution_monitor.ipp:1318
    #23 0x7ff6f1c1efdb in boost::execution_monitor::vexecute E:\.conan\2d8b53\1\src\boost\test\impl\execution_monitor.ipp:1414
    #24 0x7ff6f1c21ca7 in boost::unit_test::unit_test_monitor_t::execute_and_translate E:\.conan\2d8b53\1\src\boost\test\impl\unit_test_monitor.ipp:49
    #25 0x7ff6f1ba2dcd in boost::unit_test::framework::run E:\.conan\2d8b53\1\src\boost\test\impl\framework.ipp:1667
    #26 0x7ff6f1bcc11c in boost::unit_test::unit_test_main E:\.conan\2d8b53\1\src\boost\test\impl\unit_test_main.ipp:250
    #27 0x7ff6f1bcc24b in main E:\.conan\2d8b53\1\src\boost\test\impl\unit_test_main.ipp:306
    #28 0x7ff6f1b7cb78 in invoke_main D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
    #29 0x7ff6f1b7cacd in __scrt_common_main_seh D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
    #30 0x7ff6f1b7c98d in __scrt_common_main D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:330
    #31 0x7ff6f1b7cbed in mainCRTStartup D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_main.cpp:16
    #32 0x7ffed7127343 in BaseThreadInitThunk+0x13 (C:\WINDOWS\System32\KERNEL32.DLL+0x180017343)
    #33 0x7ffed83226b0 in RtlUserThreadStart+0x20 (C:\WINDOWS\SYSTEM32\ntdll.dll+0x1800526b0)

Address 0x00dd55afdaa0 is located in stack of thread T0 at offset 608 in frame
    #0 0x7ff6ef0cc15f in tracing::InitTracer C:\Users\INT005702\master\Provider\AccountProvider\Tracer_Common.cpp:10

  This frame has 15 object(s):
    [32, 120) 'resource_attributes'
    [48, 128) 'compiler temporary'
    [64, 192) 'resource'
    [80, 120) 'compiler temporary'
    [96, 184) 'opts'
    [112, 128) 'compiler temporary'
    [128, 136) 'exporter'
    [144, 152) 'processor'
    [160, 192) 'processors'
    [176, 184) 'context'
    [192, 208) 'provider'
    [208, 216) 'compiler temporary'
    [224, 232) 'compiler temporary'
    [240, 272) 'compiler temporary'
    [256, 272) 'compiler temporary' <== Memory access at offset 608 overflows this variable
HINT: this may be a false positive if your program uses some custom stack unwind mechanism, swapcontext or vfork
      (longjmp, SEH and C++ exceptions *are* supported)
SUMMARY: AddressSanitizer: stack-buffer-overflow C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.34.31933\include\xmemory:1080 in std::_Container_base12::_Container_base12
Shadow bytes around the buggy address:
  0x02b311d5fb00: 00 00 00 00 00 00 00 00 f1 f1 f1 f1 00 00 00 00
  0x02b311d5fb10: 00 00 00 00 00 00 00 f2 f2 f2 f2 f2 f8 f8 f8 f8
  0x02b311d5fb20: f8 f8 f8 f8 f8 f8 f2 f2 f2 f2 00 00 00 00 00 00
  0x02b311d5fb30: 00 00 00 00 00 00 00 00 00 00 f2 f2 f2 f2 f8 f8
  0x02b311d5fb40: f8 f8 f8 f2 f2 f2 f2 f2 00 00 00 00 00 00 00 00
=>0x02b311d5fb50: 00 00 00 f2[f2]f2 f2 f2 f8 f8 f2 f2 f2 f2 00 f2
  0x02b311d5fb60: 00 f2 00 00 00 00 f2 f2 f2 f2 00 f2 00 00 f2 f2
  0x02b311d5fb70: f2 f2 00 f2 00 f2 00 00 00 00 f2 f2 f2 f2 00 00
  0x02b311d5fb80: f3 f3 f3 f3 00 00 00 00 00 00 00 00 00 00 00 00
  0x02b311d5fb90: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x02b311d5fba0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
  Shadow gap:              cc
`

**Reply by sauravslr** [reply]
the memory overflow comes from the otlphttpexporteroptions class. if i dont initialise the class in inittracer function mentioned above and dont export the spans to Jaeger UI, everything seems to work fine. let me know if i can do anything to resolve this issue. Also a side note, i am using conan to consume it. but that doesnt seem to be the issue here
